### PR TITLE
fix deadloop bug

### PR DESCRIPTION
**Pull request by johnlanni**
**What this PR does**:

Fixed an deadloop error that would occur when a zk connection was closed.

deadloop profile:

image

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: database/kv/zk/client.go [hunk 1]
```diff
@@ -23,9 +23,7 @@ import (
 	"sync"
 	"sync/atomic"
 	"time"
-)
```
[the remark this answers is not in the document]
Comment (johnlanni) [reply database/kv/zk/client.go]: done

**Comment by wongoo** [comment]
@johnlanni how deadloop happens if z.Session is close?

**Comment by johnlanni** [comment]
> @johnlanni how deadloop happens if z.Session is close?

https://github.com/dubbogo/gost/pull/102/files#diff-49495334fd65db9a2e367a5fb57fd212f559a2b9082d29a826cda55c14ce85aaL223

This channel will be closed here:
https://github.com/dubbogo/go-zookeeper/blob/f9d2183d89d5f736496b45dfaae0cb7cb4f687cc/zk/conn.go#L233

and then this select will be a deadloop since the channel  will always return with `ok=false`

**Review by Mulavar — APPROVED** [review]

**Comment by wongoo** [comment]
@johnlanni For the code `case event = <-z.Session:`, if z.Session is closed, event is nil, `switch event.State` seems will panic.

**Comment by johnlanni** [comment]
> @johnlanni For the code `case event = <-z.Session:`, if z.Session is closed, event is nil, `switch event.State` seems will panic.

event is not nil, it will be an empty struct of `zk.Event`, and the `event.State` is zero by default which means `StateDisconnected`

**Comment by AlexStocks** [comment]
@wongoo wait for your approve.

**Review by wongoo — APPROVED** [review]

**Comment by wongoo** [comment]
> > @johnlanni For the code `case event = <-z.Session:`, if z.Session is closed, event is nil, `switch event.State` seems will panic.
> 
> event is not nil, it will be an empty struct of `zk.Event`, and the `event.State` is zero by default which means `StateDisconnected`

@johnlanni I have approved this PR.  But I wonder whether should the `Session  <-chan zk.Event` be defined  as a reference channel, like `<-chan *zk.Event`,  to avoid memory copy. We can discuss it in another issue.

**Review by AlexStocks — APPROVED** [review]

**Comment by AlexStocks** [comment]
> > > @johnlanni For the code `case event = <-z.Session:`, if z.Session is closed, event is nil, `switch event.State` seems will panic.
> > 
> > 
> > event is not nil, it will be an empty struct of `zk.Event`, and the `event.State` is zero by default which means `StateDisconnected`
> 
> @johnlanni I have approved this PR. But I wonder whether should the `Session <-chan zk.Event` be defined as a reference channel, like `<-chan *zk.Event`, to avoid memory copy. We can discuss it in another issue.

@johnlanni I have merge your pr. pls open another issue and answer @wongoo 's question.

**Pull request merged**